02.calendar/calendar.rb

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'date'
+require 'optparse'
+
+today = Date.today
+
+year = today.year
+month = today.month
+
+opt = OptionParser.new
+
+opt.on('-m [month]', Integer) do |m|
+  unless (1..12).cover?(m)
+    abort "cal: month '#{m}' not in range 1..12"
+  end
+  month = m
+end
+
+opt.on('-y [year]', Integer) do |y|
+  unless (1..9999).cover?(y)
+    abort "cal: year '#{y}' not in range 1..9999"
+  end
+  year = y
+end
+
+opt.parse(ARGV)
+
+beginning_of_month = Date.new(year, month, 1)
+end_of_month = beginning_of_month.next_month.prev_day
+
+puts "      #{month.to_s.rjust(2)}月 #{year}"
+puts '日 月 火 水 木 金 土'
+
+lines = "   " * beginning_of_month.wday
+(beginning_of_month..end_of_month).each do |date|
+  if today == date
+    lines += "\e[30;47m#{date.day}\e[0m"
+  else
+    lines += date.day.to_s.rjust(2)
+  end
+
+  if date.saturday?
+    lines += "\n"
+  else
+    lines += ' '
+  end
+end
+
+lines += "\n" if !end_of_month.saturday?
```

Comment:
```suggestion
lines += "\n" unless end_of_month.saturday?
```

こういう場合はunlessを使うケースが多いかも。